src/django/api/serializers.py

```diff
@@ -373,6 +375,81 @@ def get_country_name(self, facility):
         return COUNTRY_NAMES.get(facility.country_code, '')
 
 
+class FacilityClaimSerializer(ModelSerializer):
+    facility_name = SerializerMethodField()
+    oar_id = SerializerMethodField()
+    contributor_name = SerializerMethodField()
+    contributor_id = SerializerMethodField()
+
+    class Meta:
+        model = FacilityClaim
+        fields = ('id', 'created_at', 'updated_at', 'contributor_id', 'oar_id',
+                  'contributor_name', 'facility_name', 'status')
+
+    def get_facility_name(self, claim):
+        return claim.facility.name
+
+    def get_oar_id(self, claim):
+        return claim.facility_id
+
+    def get_contributor_name(self, claim):
+        return claim.contributor.name
+
+    def get_contributor_id(self, claim):
+        return claim.contributor.admin.id
+
+
+class FacilityClaimDetailsSerializer(ModelSerializer):
+    contributor = SerializerMethodField()
+    facility = SerializerMethodField()
+    status_change = SerializerMethodField()
+    notes = SerializerMethodField()
+
+    class Meta:
+        model = FacilityClaim
+        fields = ('id', 'created_at', 'updated_at', 'contact_person', 'email',
+                  'phone_number', 'company_name', 'website',
+                  'facility_description', 'preferred_contact_method', 'status',
+                  'contributor', 'facility', 'verification_method',
+                  'status_change', 'notes')
+
+    def get_contributor(self, claim):
+        return UserProfileSerializer(claim.contributor.admin).data
+
+    def get_facility(self, claim):
+        return FacilitySerializer(claim.facility).data
+
+    def get_status_change(self, claim):
+        if claim.status == FacilityClaim.PENDING:
+            return {
+                'status_change_by': None,
+                'status_change_date': None,
+                'status_change_reason': None,
+            }
+
+        return {
+            'status_change_by': claim.status_change_by.email,
```

Comment:
I hit an exception on this line.

```
ERROR:django.request:Internal Server Error: /api/facility-claims/1/deny/
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 35, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/viewsets.py", line 116, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/usr/local/src/api/views.py", line 1411, in deny_claim
    response_data = FacilityClaimDetailsSerializer(claim).data
  File "/usr/local/lib/python3.7/site-packages/rest_framework/serializers.py", line 563, in data
    ret = super(Serializer, self).data
  File "/usr/local/lib/python3.7/site-packages/rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/serializers.py", line 530, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/fields.py", line 1889, in to_representation
    return method(value)
  File "/usr/local/src/api/serializers.py", line 431, in get_status_change
    'status_change_by': claim.status_change_by.email,
AttributeError: 'NoneType' object has no attribute 'email'
```

It looks like it is caused by the fact that the `deny_claim` view is not setting the `status_change_*` fields.

Reply:
I fixed this in 204d9548f31cde4d35f74618dc355f6746b7288a